include/zephyr/drivers/led.h

```diff
@@ -141,13 +141,12 @@ __syscall int led_blink(const struct device *dev, uint32_t led,
 static inline int z_impl_led_blink(const struct device *dev, uint32_t led,
 				   uint32_t delay_on, uint32_t delay_off)
 {
-	const struct led_driver_api *api =
-		(const struct led_driver_api *)dev->api;
+	__ASSERT_NO_MSG(DEVICE_API_IS(led, dev));
 
-	if (api->blink == NULL) {
+	if (DEVICE_API_GET(led, dev)->blink == NULL) {
 		return -ENOSYS;
 	}
-	return api->blink(dev, led, delay_on, delay_off);
+	return DEVICE_API_GET(led, dev)->blink(dev, led, delay_on, delay_off);
```

Comment:
suggestion: this notation is quite heavy. Since we are using it several times, couldn't we just get the api pointer ?